### PR TITLE
capability to create first free subnet

### PR DESCRIPTION
**Pull request by tomasherout**
Hello,

phpIPAM has capability to create _Creates new child subnet inside subnet with specified mask_ -  by API call `/api/my_app/subnets/{id}/first_subnet/{mask}/` - https://phpipam.net/api/api_documentation/

phpipam-sdk-go library also supports this since 3 months ago https://github.com/pavel-z1/phpipam-sdk-go/blob/master/controllers/subnets/subnets.go#L121

My fork tries to implement it together. Subnet can be created in current way (specify subnet address and mask):

```terraform
resource "phpipam_subnet" "s1" {
  section_id = data.phpipam_section.section.section_id
  subnet_address = "10.0.0.0"
  subnet_mask = 24
}
```
or in new way by specifing new **parent_subnet_id** option:

```terraform
resource "phpipam_subnet" "s1" {
  section_id = data.phpipam_section.section.section_id
  parent_subnet_id = 12
  subnet_mask = 24
}
```

Thanks for watching it and I hope it can be merged :)

### CODE REVIEW

**Comment by pavel-z1** [comment]
Hi @tomasherout ,
Please remove from commit files: go.mod, go.sum.

We have plans to update modules version shortly. They will conflict with modules versions from this PR.

File: plugin/providers/phpipam/resource_phpipam_subnet.go [hunk 1]
```diff
@@ -3,7 +3,9 @@ package phpipam
 import (
 	"errors"
 	"fmt"
+	"github.com/pavel-z1/phpipam-sdk-go/phpipam"
```
Comment (pavel-z1) [inline plugin/providers/phpipam/resource_phpipam_subnet.go:6]: I guess that in here you can not indicate the dependence from phpipam-sdk-go.
Try to remove this line please.

**Review by pavel-z1 — CHANGES_REQUESTED** [review]

**Comment by lord-kyron** [comment]
@pavel-z1  - does it look right to you now?

**Comment by pavel-z1** [comment]
Hi @lord-kyron , it's fine for me.

**Review by pavel-z1 — APPROVED** [review]

**Comment by pavel-z1** [comment]
Thank you @tomasherout for help.

**Pull request merged**